### ISSUE TITLE
Wrong return type for socket ::send method

### ISSUE THREAD

**Issue by sxleixer** [sommer/veins#67]
You are using `size_t` as return type for socket `::send` method.

The documentation for [`send`](http://pubs.opengroup.org/onlinepubs/009696899/functions/send.html) says:

> Upon successful completion, send() shall return the number of bytes sent. Otherwise, -1 shall be returned and errno set to indicate the error.

[`size_t`](http://www.cplusplus.com/reference/cstring/size_t/) is in any occasion an `unsigned integer` type. Therefore `ssize_t` type should be used as provided by `sys/socket.h`.

The following patch corrects this issue:
```patch
Index: src/veins/modules/mobility/traci/TraCIConnection.cc
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/veins/modules/mobility/traci/TraCIConnection.cc	(date 1527027511000)
+++ src/veins/modules/mobility/traci/TraCIConnection.cc	(date 1527764527000)
@@ -184,7 +184,7 @@
 		buf2 << msgLength;
 		uint32_t bytesWritten = 0;
 		while (bytesWritten < sizeof(uint32_t)) {
-			size_t sentBytes = ::send(socket(socketPtr), buf2.str().c_str() + bytesWritten, sizeof(uint32_t) - bytesWritten, 0);
+			ssize_t sentBytes = ::send(socket(socketPtr), buf2.str().c_str() + bytesWritten, sizeof(uint32_t) - bytesWritten, 0);
 			if (sentBytes > 0) {
 				bytesWritten += sentBytes;
 			} else {
@@ -199,7 +199,7 @@
 		MYDEBUG << "Writing TraCI message of " << buf.length() << " bytes" << endl;
 		uint32_t bytesWritten = 0;
 		while (bytesWritten < buf.length()) {
-			size_t sentBytes = ::send(socket(socketPtr), buf.c_str() + bytesWritten, buf.length() - bytesWritten, 0);
+			ssize_t sentBytes = ::send(socket(socketPtr), buf.c_str() + bytesWritten, buf.length() - bytesWritten, 0);
 			if (sentBytes > 0) {
 				bytesWritten += sentBytes;
 			} else {
```

**Comment by sxleixer** [comment]
Additionally the `else` branch was only taken if `sentBytes = 0` and never when `sentBytes = -1 = 18446744073709551615` as we would expect for error handling. I received a `SIGPIPE` upon that problem.

**Comment by sommer** [comment]
Merged into b0f76a93
